src/os/bluestore/BlueStore.cc

```diff
@@ -8174,7 +8174,16 @@ int BlueStore::statfs(struct store_statfs_t *buf,
 int BlueStore::pool_statfs(uint64_t pool_id, struct store_statfs_t *buf)
 {
   dout(20) << __func__ << " pool " << pool_id<< dendl;
+
   if (!per_pool_stat_collection) {
+    if (cct->_conf->bluestore_no_per_pool_stats_tolerance != "enforce" &&
+      cct->_conf->bluestore_warn_on_legacy_statfs) {
+      string s("legacy statfs reporting detected, "
+	"suggest to run store repair to get consistent statistic reports");
+      _set_legacy_statfs_alert(s);
+    } else {
+      _clear_legacy_statfs_alert();
+    }
```

Comment:
it seems like this should go in mount() or similar, not every pool_statfs() call?

Reply:
actually, we should put it in a helper that raises alerts, and re-call the helper from handle_conf_change(), and add the config option on the watched list, so the user can turn off the warnings without restarting osds.